kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/SchemaManagerTest.java

```diff
@@ -66,5 +68,80 @@ public void testBQTableDescription() {
 
     Assert.assertEquals("Kafka doc does not match BigQuery table description",
                         testDoc, tableInfo.getDescription());
+
+    Assert.assertNull("Timestamp partition field name is not null",
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getField());
+  }
+
+  @Test
+  public void testTimestampPartitionSet() {
+    final String testTableName = "testTable";
+    final String testDatasetName = "testDataset";
+    final String testDoc = "test doc";
+    final String testField = "testFieldName";
+    final TableId tableId = TableId.of(testDatasetName, testTableName);
+
+    SchemaRetriever mockSchemaRetriever = mock(SchemaRetriever.class);
+    @SuppressWarnings("unchecked")
+    SchemaConverter<com.google.cloud.bigquery.Schema> mockSchemaConverter =
+        (SchemaConverter<com.google.cloud.bigquery.Schema>) mock(SchemaConverter.class);
+    BigQuery mockBigQuery = mock(BigQuery.class);
+
+    SchemaManager schemaManager = new SchemaManager(mockSchemaRetriever,
+        mockSchemaConverter,
+        mockBigQuery,
+        testField);
+
+    Schema mockKafkaSchema = mock(Schema.class);
+    // we would prefer to mock this class, but it is final.
+    com.google.cloud.bigquery.Schema fakeBigQuerySchema =
+        com.google.cloud.bigquery.Schema.of(Field.of("mock field", LegacySQLTypeName.STRING));
+
+    when(mockSchemaConverter.convertSchema(mockKafkaSchema)).thenReturn(fakeBigQuerySchema);
+    when(mockKafkaSchema.doc()).thenReturn(testDoc);
+
+    TableInfo tableInfo = schemaManager.constructTableInfo(tableId, mockKafkaSchema);
+
+    Assert.assertEquals("Kafka doc does not match BigQuery table description",
+        testDoc, tableInfo.getDescription());
+
+    Assert.assertEquals( "The field name does not match the field name of time partition",
+        testField,
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getField());
+  }
+
+  @Test
+  public void testNullTimestampPartitionName() {
+    final String testTableName = "testTable";
+    final String testDatasetName = "testDataset";
+    final String testDoc = "test doc";
+    final TableId tableId = TableId.of(testDatasetName, testTableName);
+
+    SchemaRetriever mockSchemaRetriever = mock(SchemaRetriever.class);
+    @SuppressWarnings("unchecked")
+    SchemaConverter<com.google.cloud.bigquery.Schema> mockSchemaConverter =
+        (SchemaConverter<com.google.cloud.bigquery.Schema>) mock(SchemaConverter.class);
+    BigQuery mockBigQuery = mock(BigQuery.class);
+
+    SchemaManager schemaManager = new SchemaManager(mockSchemaRetriever,
+        mockSchemaConverter,
+        mockBigQuery,
+        null);
```

Comment:
ok i know youre not this bad at formatting, does wepay have some different standard for formatting? this looks gross